homeassistant/components/tplink/sensor.py

```diff
@@ -154,13 +156,20 @@ def __init__(
         device: SmartDevice,
         coordinator: TPLinkDataUpdateCoordinator,
         description: TPLinkSensorEntityDescription,
+        has_parent: bool = False,
     ) -> None:
         """Initialize the switch."""
         super().__init__(device, coordinator)
         self.entity_description = description
-        self._attr_unique_id = (
-            f"{legacy_device_id(self.device)}_{self.entity_description.key}"
-        )
+        self._attr_unique_id = f"{legacy_device_id(device)}_{description.key}"
+        if has_parent:
+            assert device.alias
```

Comment:
`device.alias` is `str | None`, when can it be None? With what should we replace it when this is the case?

Reply:
I think it could be set to unknown, but this should not actually ever be None in homeassistant's context, right @sdb9696?

Reply:
I guess I have a good test case for this since I haven't labeled the plugs...let me give it a shot